File: src/main/angular/src/app/model/route-detail-move.ts
/**
 * ルート詳細移動
 */
export interface IRouteDetailMove {
  /** ルート詳細ID */
  routeDetailId?: number;
  /** 所要時間 */
  moveMinutes?: number;
  /** 移動種類区分 */
  moveWayDiv?: number;
  /** 移動種類名称 */
  moveWayDivName?: string;
  /** 移動費用 */
  moveCost?: number;
}

File: src/main/kotlin/net/yttravelers/spotshare/entities/RouteDetailMove.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Table
import javax.persistence.*

@Entity
@Table(name = "routeDetailMove")
class RouteDetailMove {
    @Id
    var routeDetailId: Int? = null

    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "routeDetailId")
    @MapsId
    lateinit var routeDetail: RouteDetail

    var moveMinutes: Int? = null

    //TODO: 区分の設計
//    var moveKindDiv:
    //必要？
//    var moveKindDivName:

    var moveCost: Int? = null

}